### DISCUSSION TITLE
How do I coerce types in my schema?

### DISCUSSION THREAD

**Discussion by ScottChapman** [hasura/graphql-engine#9690] [Question] [answered]
I appologize for the noob question, I have poked around the docs and googled a bit, but am not really sure where II should be looking.

I have a postgresql database, and unfortunately the fields that are timestamps are actually stored as ints (unix time stamps).
I'd like to provide marshaling int <-> timestamp in Hasura to insulate API users from that database design decision.

**Comment by SamirTalwar** [comment, marked as answer]
You can probably solve this when reading the data by creating a function to do the conversion, and then using a [computed field](https://hasura.io/docs/latest/schema/postgres/computed-fields/).

Alternatively, you can write a [native query](https://hasura.io/docs/latest/schema/postgres/logical-models/native-queries/) to convert the data any way you like. A future release of Hasura will also support mutations using native queries.

Does that help?

**Reply by ScottChapman** [reply]
Thanks, that's helpful. I think sometimes we don't have control over the schema of the database, and it is a useful pattern to be able to address that in one place on behalf of clients.

I've seen other systems provide custom marshaling of data to support to/from the native type in a very flexible manner; ability to provide custom code to convert to the standard types, or to introduce custom scalars
